### ISSUE TITLE
[kws_streaming]  CRNN model less streaming accuracy

### ISSUE THREAD

**Issue by vishalkevat007** [google-research/google-research#819]
@rybakov I have trained the CRNN model as specified in the recipe [here](https://github.com/google-research/google-research/blob/master/kws_streaming/experiments/kws_experiments_paper_12_labels.md#crnn). with training steps 20000,20000

I have got the a good testing accuracy of 97.23% for Non-Streaming model ,  but for the streaming case, it very less about 53.67%  for the both models( internal and external states), I couldn't find what was the issue. Can you please tell what went wrong and how to improve the streaming accuracy?

**Comment by rybakov** [comment]
This effect is well described in our [paper](https://arxiv.org/pdf/2005.06720.pdf), please have a look at paragraph "4.3. Streaming and non-streaming latency with accuracy". If you train RNN based model with default settings (CRNN is based on RNN) then it resets RNN state at every minibatch and trained in "stateless" mode - this model can not be used for streaming, because it expects that RNN state is reseted in the beginning of the input sequence, but in the streaming mode we do not know when is the beginning of the key word in the input audio sequence. That is why for streaming mode, RNN based model, have to be trained in "statefull" mode as shown in [crnn_state](https://github.com/google-research/google-research/blob/master/kws_streaming/experiments/kws_experiments_paper_12_labels.md#crnn_state). In this case you should expect that overall accuracy of CRNN model can go down: as shown in "Table 2: Accuracy on data V2 with latency and model size" in comparison to non streaming version in "Table 1: Baseline models accuracy on data V1* and V2* with
paper references and our models accuracy on data V1 and V2" of the paper. Model has to learn on its own how to process RNN state in streaming mode.

Non streaming accuracy reported at "Table 1: Baseline models accuracy on data V1* and V2* with
paper references and our models accuracy on data V1 and V2" can be reproduced by running [crnn](https://github.com/google-research/google-research/blob/master/kws_streaming/experiments/kws_experiments_paper_12_labels.md#crnn)

Streaming accuracy reported at "Table 2: Accuracy on data V2 with latency and model size" can be reproduced by [crnn_state](https://github.com/google-research/google-research/blob/master/kws_streaming/experiments/kws_experiments_paper_12_labels.md#crnn_state)

**Comment by vishalkevat007** [comment]
Ohkay, understood. 
Thankyou @rybakov
